src/supernova/mod.rs

```diff
@@ -794,18 +798,14 @@ where
         }
       })?;
 
-    self.r_U_secondary[0].as_ref().map_or(Ok(()), |U| {
-      if U.X.len() != 2 {
-        debug!(
-          "r_U_secondary got instance length {:?} != {:?}",
-          U.X.len(),
-          2
-        );
-        Err(SuperNovaError::NovaError(NovaError::ProofVerifyError))
-      } else {
-        Ok(())
-      }
-    })?;
+    if self.r_U_secondary.X.len() != 2 {
+      debug!(
+        "r_U_secondary got instance length {:?} != {:?}",
+        self.r_U_secondary.X.len(),
+        2
+      );
+      Err(SuperNovaError::NovaError(NovaError::ProofVerifyError))?
```

Comment:
I think this is missing an important `return` statement.

Reply:
I thought the `?` at the end would work too, but the return is more clear.